File: scripts/aggregation1.js
var connection = new Mongo();
var db = connection.getDB('Crimes');

var result = db.list.aggregate([
	{ $match: {"Month": "2012-06"} }, 
	{ $group: {_id: "$Crime type", "total": {$sum: 1}} }, 
	{ $sort: {"total": -1} }, { $limit : 5}
]);

printjson(result);

File: scripts/aggregation1.py
import pymongo
from pymongo import MongoClient
from bson.son import SON

client = MongoClient('localhost', 27017)
db = client.Crimes
collection = db.list

pipeline = [
	{ "$match": {"Month": "2012-06"} }, 
	{ "$group": {"_id": "$Crime type", "total": {"$sum": 1}} }, 
  	{ "$sort": SON([("total", -1)])},
	{ "$limit": 5 }
]

result = collection.aggregate(pipeline)

for doc in result:
	print doc
